libfreerdp/codec/region.c

```diff
@@ -132,7 +132,10 @@ static RECTANGLE_16 *region16_extents_noconst(REGION16 *region)
 
 BOOL rectangle_is_empty(const RECTANGLE_16 *rect)
 {
-	return (rect->left + rect->top + rect->right + rect->bottom) ? TRUE : FALSE;
+    /* An rectangle with width <= 0 or height <= 0 should be regarded
+     * as empty.
+     */
```

Comment:
I agree an invalid rectangle should be denied.
Actually my original idea is just test "right = left or bottom = top". If we don't like width <0 and height < 0, we can change to just check '='.
Or do you think we should be more strict, only allow (0,0,0,0)? I think that's what the original code thinks, but it returns true when it is NOT empty.